src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -1121,8 +1121,9 @@ traditionalStatement
         // up, but that's pretty hard without a symbol table ;)
         |    (declaration)=> declaration SEMI
 
-        // record declaration, note that you cannot have modifiers in this case
-        |   recordDefinition[#null]
+        // record declaration, note that you cannot have modifiers in this case so
+        // we create an empty modifiers AST to avoid NPE
```

Comment:
`// we create an empty modifiers AST as we do for classes without modifiers`

Reply:
Fixed.